src/EventListener/InvalidateRequestSubscriber.php

```diff
@@ -37,8 +44,48 @@ public static function getSubscribedEvents() : array
      */
     public function invalidate(InvalidateRequestEvent $event)
     {
-        $cacheStorage = $this->cacheMiddleware->getCacheStorage();
-        $requests     = $event->getRequests();
-        array_walk($requests, [$cacheStorage, 'delete']);
+        $client = $event->getClient();
+        $objectIdentifier = $this->getClientIdentifier($client);
+
+        if (isset($this->cacheMiddlewares[$objectIdentifier])) {
+            $cacheStorage = $this->cacheMiddlewares[$objectIdentifier]->getCacheStorage();
+
+            $request = $this->buildRequest(
+                $event->getMethod(),
+                $client->getConfig('base_uri'),
+                $event->getUri()
+            );
+
+            $cacheStorage->delete($request);
+        }
+    }
+
+    /**
+     * @param string $method
+     * @param UriInterface|null $baseUri
+     * @param string $requestUri
+     *
+     * @return Request
+     */
+    protected function buildRequest(string $method, $baseUri, string $requestUri) : Request
```

Comment:
I would prefer it in the Event object. It is available only in this class but we coul need it in another event subscriber after invalidate or before... also the cache key is constructed using the request object so it would make sense...

Reply:
I don't prefer to have logic in event class but ok, it makes sense. Moves one method to event class :+1: